### PR TITLE
added stage to debug gorelease failure

### PR DESCRIPTION
**Pull request by A-kanksh-a**
## Change Overview
This PR is to add a test job in kanister to build all tools images before the release. This will let us avoid image build failures when doing a kanister release.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #K10-6428

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -127,8 +127,6 @@ dockers:
   dockerfile: 'docker/kafka-adobes3Connector/image/adobeSink.Dockerfile'
   extra_files:
   - 'docker/kafka-adobes3Connector/image/adobe-monitorsink.sh'
-archives:
-- allow_different_binary_count: true
```
Comment (PrasadG193) [inline .goreleaser.yml:131]: Why this change is required @A-kanksh-a ?

Comment (A-kanksh-a) [reply .goreleaser.yml:131]: I had added this change as a part of [https://github.com/kanisterio/kanister/pull/840](https://github.com/kanisterio/kanister/pull/840) as there was some difference in binary count while building `kafka adobe s3 connector` image on the local environment.
After I added `make release-snapshot` command, the kanister CI job failed and was causing an issue with this field `allow_different_binary_count` set to true. 
Also on Kanister CI environment building `kafka adobe s3 connector` image is not causing issue with difference in binary count.

Comment (A-kanksh-a) [reply .goreleaser.yml:131]: I had added this as a part of [https://github.com/kanisterio/kanister/pull/840](https://github.com/kanisterio/kanister/pull/840), as I was getting different binary counts issue on the local machine while adding `kafka-adobe-s3-connector` related image to goreleaser.

After adding the command `make release-snapshot` with field `allow_different_binary_count` set to `true` caused issue.

And also, Kanister CI environment is not having different binary counts issue for `kafka-adobe-s3-connector` related image, so that flag is not needed.

**Review by pavannd1 — APPROVED** [review]
